service/src/main/java/uk/gov/hmcts/reform/fpl/json/serializer/ChildSelectorSerializer.java

```diff
@@ -0,0 +1,39 @@
+package uk.gov.hmcts.reform.fpl.json.serializer;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import org.springframework.boot.jackson.JsonComponent;
+import uk.gov.hmcts.reform.fpl.model.order.generated.selector.ChildSelector;
+
+import java.io.IOException;
+import java.util.List;
+
+@JsonComponent
+public class ChildSelectorSerializer extends JsonSerializer<ChildSelector> {
+    @Override
+    public void serialize(ChildSelector value, JsonGenerator gen,
+                          SerializerProvider serializers) throws IOException {
+        gen.writeStartObject();
```

Comment:
data is throw away, why do we need to serialise again? I would have thought the flow would be

Get selected children, find their indexes. Get children with indexes. Put where ever name is needed.

Reply:
Deserialiser is useful I think but the serialiser can be removed, at the moment we don't need to really serialise an object can just do
```java
caseDetails.getData.().put("childSelector", Map.of("childCountContainer", string_value));
```

Reply:
Actually, no the serialiser will be needed in the mid events

Reply:
Removing the serialiser broke the tests 😞

Reply:
Removed serialiser and deserialiser